Add launching mode and intake

CodingKoopa opened the pull request:
This PR:
- Adds a dedicated launching mode for the manipulator, in which the bot will autonomously adjust itself to make the shot into the hoop, vertically. Ensuring that the shot can be made horizontally will make this more complicated, but that can be addressed in a later pull request. 
- Adds D-pad inputs for the control panel. This only loosely relates to the main focus of this PR, but it is grouped together because of how I restructured some of the inputs in order to make launching mode flow well.
- Adds preliminary (very untested) code for using the photosensors to keep track of the balls.
- Adds Shuffleboard telemetry for keeping track of the ball status.
- Spins up the launching motor when there are enough balls in storage.

CodingKoopa committed:
Add more motor ports.

Commit:
Add D-pad and back input presses.

Commit:
Change control panel spin amount to the D-pad.

Commit:
Add launching mode.

Commit:
Add preliminary ball intake system.

Commit:
Allow manual movement of belt.

Commit:
Allow manual starting of launcher motor.

Commit:
Uncomment experimental adjustment.

RyanStyron commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -69,6 +70,19 @@
   private final DifferentialDrive m_differentialDrive =
       new DifferentialDrive(m_motorDriveFrontLeft, m_motorDriveFrontRight);
 
+  // Ball Intake
```
Is there a reason for why certain variables in this section, including `photoelectricSensorEnter`, `ballDetectedEnterLastLoop`, etc., are not specified as member variables?

CodingKoopa replied:
Nope. Fixed.

RyanStyron reviewed (commented):
With the exception of my previous comment on this PR and merging this with recent commits, predominantly those regarding Shuffleboard, it all looks good to me. Very nice work.

RyanStyron commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -440,57 +514,134 @@ private void driveSpeed() {
   /**
-   * Determines whether or not the ball can be launched into the power port, and adjusts the robot to
-   * make the shot if it can't.
+   * This function determines whether or not the ball can be launched into the power port, and adjusts
```
This is very nit-picky, however in terms of consistency purposes, I believe it would be appropriate to remove "This function."

Commit:
Move Shuffleboard stuff to ShuffleboardHelper.

Commit:
Add spacing to ShuffleboardHelper.

Commit:
Lower default tolerance value.

Commit:
Increase constant of proportionality.

Commit:
Move more state into HandleState.

Commit:
Fix photosensor ports.

CodingKoopa committed:
Fix button ID names.

CodingKoopa committed:
Add missing lastLoop setting.

CodingKoopa committed:
Disallow driving while in launching mode.

This is done as to not set the drive motor powers more than once in a
loop.

CodingKoopa committed:
Put a cap on the correctional drive, and disable square.

CodingKoopa committed:
Fix member variable names.

CodingKoopa committed:
Move initialization to disabledInit.

CodingKoopa committed:
Use variable when setting ShuffleBoard for consistency.

CodingKoopa committed:
Oops! All member variables

CodingKoopa commented:
Do not merge - will depend on another PR.

CodingKoopa committed:
Order balls in storage first, for consistency with Shuffleboard.

CodingKoopa committed:
Update ShuffleboardHelper comment.

CodingKoopa committed:
Add initial setting of Shuffleboard values

CodingKoopa committed:
Disallow overlap of states when toggled from Shuffleboard.

Previously, the overlap check would only affect the state when checking
with buttons.

CodingKoopa committed:
Manually set the launching mode entry, as pressing back would.

CodingKoopa committed:
Uninvert the launching motor.

The testing of Robot2020MotorTest showed that this isn't necessary.

CodingKoopa committed:
Fix class of right launcher motor.

CodingKoopa committed:
Run launcher at full speed, as specified by Lang.

CodingKoopa committed:
Update launchBall value to false if not within tolerance.

Without this, if we are within the range and then leave it, the robot
code will have the false impression that it can still make the shot.

CodingKoopa committed:
Add 0 spin control, for testing and emergencies.

CodingKoopa committed:
Clean up Shuffleboard.

This commit marks sections whose values are not checked by the robot
code as read only. Ball launching status has been moved to the
launching layout. Some labels have been shortened to fit better.

CodingKoopa committed:
Default distance tolerance to 0.25.

CodingKoopa commented:
Pending on #34.

RyanStyron reviewed (approved):
LGTM, nice work.

CodingKoopa committed:
Fix variable name.
